### PR TITLE
Add list buckets API to S3

### LINKED ISSUE #6333: [Feature] Add API to fetch all the buckets from S3 datasource
To generate a page from the DB table, to select the bucket on which the CRUD page needs to be created create an API to fetch all the buckets from the connected S3 datasource.

### PR DESCRIPTION
**Pull request by abhvsn**
To generate a page from the DB table, to select the bucket on which the CRUD page needs to be created create an API to fetch all the buckets from the connected S3 datasource.

Fixes #6333

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Tested manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java [hunk 1]
```diff
@@ -890,5 +903,16 @@ public void datasourceDestroy(AmazonS3 connection) {
              */
             return Mono.empty();
         }
+
+        @Override
+        public Mono<ActionExecutionResult> getDatasourceMetadata(List<Property> pluginSpecifiedTemplates,
```
Comment (sumitsum) [inline app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:908]: Minor: Can you please add a comment here explaining how this function gets used ?

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java [hunk 2]
```diff
@@ -545,6 +550,14 @@ String readFile(AmazonS3 connection, String bucketName, String path, Boolean enc
                         connection.deleteObject(bucketName, path);
                         actionResult = Map.of("status", "File deleted successfully");
                         break;
+                    case LIST_BUCKETS:
```
Comment (sumitsum) [inline app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:553]: Can you please add a test for this action ?

Comment (abhvsn) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:553]: Updated code pushed.

File: app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java (hunk 1 above)
Comment (abhvsn) [reply app/server/appsmith-plugins/amazons3Plugin/src/main/java/com/external/plugins/AmazonS3Plugin.java:911]: Added a minor comment about the function use case, for more details refer pluginExecutor

**Review by sumitsum — APPROVED** [review]

**Pull request merged**